ext/standard/basic_functions.stub.php

```diff
@@ -395,3 +395,199 @@ function assert($assertion, $description = null): bool {}
 
 /** @return int|string|bool|null */
 function assert_options(int $what, $value = UNKNOWN) {}
+
+/* string.c */
+
+function bin2hex(string $data): string {}
+
+/** @return string|false */
+function hex2bin(string $data) {}
+
+/** @return int|false */
+function strspn(string $str, string $mask, int $start = UNKNOWN, int $len = UNKNOWN) {}
+
+/** @return int|false */
+function strcspn(string $str, string $mask, int $start = UNKNOWN, int $len = UNKNOWN) {}
+
+#if HAVE_NL_LANGINFO
+/** @return string|false */
+function nl_langinfo(int $item) {}
+#endif
+
+function strcoll(string $str1, string $str2): int {}
+
+function trim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function rtrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function ltrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function wordwrap(string $str, int $width = 75, string $break = "\n", bool $cut = false): string {}
+
+function explode(string $separator, string $str, int $limit = PHP_INT_MAX): array {}
+
+/**
+ * @param string $glue Optional - defaults to empty string
+ * @param array $pieces
+ */
+function implode($glue, $pieces): string {}
```

Comment:
```suggestion
function implode($glue, $pieces = UNKNOWN): string {}
```

(We need to add some sanity assertions in the engine for this...)